src/libraries/Microsoft.Extensions.Configuration.Binder/tests/ConfigurationCollectionBindingTests.cs

```diff
@@ -1602,5 +1602,68 @@ private class OptionsWithInterdependentProperties
             public IEnumerable<int> FilteredConfigValues => ConfigValues.Where(p => p > 10);
             public IEnumerable<int> ConfigValues { get; set; }
         }
+
+        [Fact]
+        public void DifferentDictionaryBindingCasesTest()
+        {
+            var dic = new Dictionary<string, string>() { { "key", "value" } };
+            var config = new ConfigurationBuilder()
+                .AddInMemoryCollection(dic)
+                .Build();
+
+            var _dictionary = config.Get<Dictionary<string, string>>();
```

Comment:
nit: the variable naming, I wouldn't have them with underscore, or if you prefer no need to define the variables since each have one statements have it inline with Assert.Single statement